### PR TITLE
chore: change hooks to be taken by ref

### PR DESCRIPTION
**Pull request by asafshen**
## Related Issues
https://github.com/descope/etc/issues/5319

## Description
change hooks to be calculated on function runtime instead on config time

will later enables taking them in a dynamic manner

### CODE REVIEW

File: packages/core-js-sdk/src/createSdk.ts [hunk 1]
```diff
@@ -30,21 +30,26 @@ const withMultipleHooks =
         beforeRequest?: BeforeRequest | BeforeRequest[];
         afterRequest?: AfterRequest | AfterRequest[];
       };
-    }
+    },
   ) => {
-    const beforeRequestHooks = [].concat(config.hooks?.beforeRequest || []);
-    const afterRequestHooks = [].concat(config.hooks?.afterRequest || []);
+    const beforeRequest: BeforeRequest = (c) => {
```
Comment (nirgur) [inline packages/core-js-sdk/src/createSdk.ts:35]: Can we change `c` to `conf`?

File: packages/core-js-sdk/test/createSdk.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const mockFetch = jest.fn();
+globalThis.fetch = mockFetch;
+
+describe('createSdk', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should call "beforeRequest" and "afterRequest"', async () => {
```
Comment (nirgur) [inline packages/core-js-sdk/test/createSdk.test.ts:11]: I would split it into 2 different tests and test that when changing the hook it's actually working
I think that we already have tests for calling hooks

**Review by nirgur — COMMENTED** [review]
Cool
Did you check it's actually working?

**Comment by asafshen** [comment]
> Cool
Did you check it's actually working?

Of Course

File: packages/core-js-sdk/test/createSdk.test.ts (hunk 2 above)
Comment (asafshen) [reply packages/core-js-sdk/test/createSdk.test.ts:11]: unfortunately we don't have tests for this code
(only with mocks of the inner object)

File: packages/core-js-sdk/src/createSdk.ts (hunk 1 above)
Comment (asafshen) [reply packages/core-js-sdk/src/createSdk.ts:35]: done

File: packages/core-js-sdk/test/createSdk.test.ts (hunk 2 above)
Comment (asafshen) [reply packages/core-js-sdk/test/createSdk.test.ts]: good idea for splitting, done 👍

Comment (nirgur) [reply packages/core-js-sdk/test/createSdk.test.ts]: Cool, thanks!
Don't you want to add a test where the hooks are updated after initialization?
just to make sure someone will not change it without noticing

Comment (asafshen) [reply packages/core-js-sdk/test/createSdk.test.ts]: good idea 👍  added such test

**Review by nirgur — APPROVED** [review]

**Pull request merged**